pkg/controllers/repositoryserver/repositoryserver_controller.go

```diff
@@ -96,6 +97,43 @@ func (r *RepositoryServerReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		r.Recorder.Event(repoServerHandler.RepositoryServer, corev1.EventTypeWarning, "Failed", err.Error())
 		return ctrl.Result{}, err
 	}
+
+	logger.Info("Connect to Kopia Repository")
+	if err := repoServerHandler.connectToKopiaRepository(); err != nil {
+		repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Failed
+		if uerr := r.Status().Update(ctx, repoServerHandler.RepositoryServer); uerr != nil {
+			return ctrl.Result{}, uerr
+		}
+		return ctrl.Result{}, err
```

Comment:
lets add an event here the way we are adding at https://github.com/kanisterio/kanister/pull/2136/files#diff-23ca9f31d9f80b5f4d59f7528867cbd53948427aa21e696cfda03e8230870578R97

Reply:
For event handling, I planned to raise another PR as there are other events as well that we might want to add, for example `success` Events. Let me know if you still think we should handle it in this PR itself.

Reply:
lets create a ticket for it then, WDYT?

Reply:
Created this issue [https://github.com/kanisterio/kanister/issues/2148](https://github.com/kanisterio/kanister/issues/2148)